src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/ProvidePublicParameterlessSafeHandleConstructor.cs

```diff
@@ -73,6 +73,13 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbo
                         return;
                     }
 
+                    if ((constructor.DeclaredAccessibility == Accessibility.ProtectedAndInternal && type.DeclaredAccessibility == Accessibility.ProtectedAndInternal)
+                        || constructor.DeclaredAccessibility == Accessibility.Internal && type.DeclaredAccessibility == Accessibility.Internal)
+                    {
+                        // The parameterless constructor is as visible as the containing type, so there is no diagnostic to emit.
```

Comment:
@mavasani, is there an existing helper that does this, that says "is this member visible outside of the type"?

Reply:
I think you probably want to use either of the below extension methods: https://github.com/dotnet/roslyn-analyzers/blob/a0522afd52b25ad211298e308fd5bbc7734865e0/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L160-L166

Reply:
I did not know those extension methods existed. I'll switch this code over to using them